spec/controllers/hq/users_controller_spec.rb

```diff
@@ -31,4 +31,74 @@ def foobar_action
     expect(response).to render_template 'index'
   end
 
+  describe '#edit and #update' do
+    before :each do
+      @old_user = FactoryGirl.build_stubbed(:user)
+      expect(User).to receive(:find).and_return(@old_user)
+    end    
+
+    specify 'editing renders the edit view' do
+      get :edit, id: @old_user.id
+      expect(response).to render_template 'edit'
+    end
+
+    it 'handles invalid updates' do
+      allow(@old_user).to receive(:save).and_return(false)
+      post :update, id: @old_user.id, user: { email: 'some email address'}
+      expect(response).to render_template 'edit'
+      new_user = controller.instance_variable_get(:@user)
+      expect(new_user[:email]).to eq 'some email address'
+      expect(flash[:success]).to be_nil
+    end
+
+    it 'handles valid updates' do
+      allow(@old_user).to receive(:save).and_return(true)
+      post :update, id: @old_user.id, user: { email: 'some email address'}
+      expect(response).to redirect_to(hq_user_path(@old_user))
+      expect(flash[:success]).to_not be_nil
+    end
+
+  end
+
+  describe '#new' do
+    
+    before :each do
+      @new_user = FactoryGirl.build_stubbed(:user, user_name: nil, email: nil)
+      allow(User).to receive(:new).and_return(@new_user)
+    end
+
+    it 'enables entering details of a new user' do
+      get :new
+      expect(response).to render_template 'new'
+      user = controller.instance_variable_get(:@user)
+      expect(user).to_not be_nil
+      expect(user[:user_name]).to be_nil
+    end
+
+  end
+
+  describe '#create' do
+
+    before :each do
+      @new_user = FactoryGirl.build_stubbed(:user, user_name: nil, email: nil)
+      allow(User).to receive(:new).and_return(@new_user)
+    end
+
+    specify 'successful create shows the new user' do
+      expect(@new_user).to receive(:save).and_return(true)
+      post :create, user: { user_name: 'some user' }
+      expect(response).to redirect_to hq_user_path(@new_user)
+      expect(flash[:success]).to_not be_nil
+    end
+
+    specify 'unsuccessful create re-renders the new view' do
+      expect(@new_user).to receive(:save).and_return(false)
+      post :create, user: { user_name: 'some user' }
+      expect(response).to render_template 'new'
+      user = controller.instance_variable_get(:@user)
```

Comment:
Setting of the instance variable is covered by the tests indirectly but thoroughly: 
- you tell the User model to return the factory-generated object when it receives a call to `find`;
- you then tell the factory object to expect a `save` message;
- and finally you verify than on successful save there is a redirection to the factory object's page

All of the above expectations can only be met if the instance variable is set and used correctly throughout the controller action. Explicitly testing that this is so, however, means testing the implementation, which is unnecessarily low-level and brittle.